track/gpsd.py

```diff
@@ -0,0 +1,185 @@
+"""GPS location determination via gpsd.
+
+Defines a single class GPS that encapsulates the procedures necessary to obtain the current location
+from the gpsd daemon. This code works on the assumption that gpsd is already configured and running
+on the system.
+"""
+
+# http://manpages.ubuntu.com/manpages/bionic/man5/gpsd_json.5.html
+
+from collections import namedtuple, OrderedDict
+from enum        import Enum, Flag, auto
+from math        import inf, nan, isnan
+from time        import perf_counter
+#from types import SimpleNamespace
+
+import gps
+
+# improved enum-ized version of the 'MODE_' constants from the gps module
+class GPSFixType(Enum):
+    ZERO   = 0
+    NO_FIX = 1
+    FIX_2D = 2
+    FIX_3D = 3
+
+GPSLocation = namedtuple('GPSLocation', ['lat', 'lon', 'alt'])
+GPSErrors   = namedtuple('GPSErrors',   ['lat', 'lon', 'alt', 'track', 'speed', 'climb', 'time'])
```

Comment:
There's also this syntax that I didn't know of prior to yesterday, which I kinda like:

```
class ModelParameters(NamedTuple):
    lon_axis_offset: Angle
    lat_axis_offset: Angle
    pole_rot_axis_lon: Angle
    pole_rot_angle: Angle
```

(the type attributes don't imply any kind of enforcement by the interpreter; they're mainly for documentation and as metadata for certain kinds of linters that do type checking)

Reply:
Okay, good to know. I'll probably go for this. :+1: